client/rest-high-level/src/test/java/org/elasticsearch/client/asyncsearch/AsyncSearchIT.java

```diff
@@ -21,37 +21,52 @@
 
 import org.elasticsearch.client.ESRestHighLevelClientTestCase;
 import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.core.AcknowledgedResponse;
 import org.elasticsearch.common.settings.Settings;
-import org.elasticsearch.common.unit.TimeValue;
 import org.elasticsearch.index.query.QueryBuilders;
 import org.elasticsearch.search.builder.SearchSourceBuilder;
 
 import java.io.IOException;
-import java.util.concurrent.TimeUnit;
 
 public class AsyncSearchIT extends ESRestHighLevelClientTestCase {
 
-    public void testSubmitAsyncSearchRequest() throws IOException {
+    public void testAsyncSearch() throws IOException {
         String index = "test-index";
         createIndex(index, Settings.EMPTY);
 
         SearchSourceBuilder sourceBuilder = new SearchSourceBuilder().query(QueryBuilders.matchAllQuery());
-        SubmitAsyncSearchRequest request = new SubmitAsyncSearchRequest(sourceBuilder, index);
-        // 15 sec should be enough to make sure we always complete right away
-        request.setWaitForCompletion(new TimeValue(15, TimeUnit.SECONDS));
-        AsyncSearchResponse response = highLevelClient().asyncSearch().submitAsyncSearch(request, RequestOptions.DEFAULT);
-        assertTrue(response.getVersion() >= 0);
-        assertFalse(response.isPartial());
-        assertTrue(response.getStartTime() > 0);
-        assertTrue(response.getExpirationTime() > 0);
-        assertNotNull(response.getSearchResponse());
-        if (response.isRunning() == false) {
-            assertNull(response.getId());
-            assertFalse(response.isPartial());
+        SubmitAsyncSearchRequest submitRequest = new SubmitAsyncSearchRequest(sourceBuilder, index);
+        submitRequest.setCleanOnCompletion(false);
+        AsyncSearchResponse submitResponse = highLevelClient().asyncSearch().submitAsyncSearch(submitRequest, RequestOptions.DEFAULT);
+        assertNotNull(submitResponse.getId());
+        assertTrue(submitResponse.getVersion() >= 0);
+        assertFalse(submitResponse.isPartial());
+        assertTrue(submitResponse.getStartTime() > 0);
+        assertTrue(submitResponse.getExpirationTime() > 0);
+        assertNotNull(submitResponse.getSearchResponse());
+        if (submitResponse.isRunning() == false) {
+            assertFalse(submitResponse.isPartial());
         } else {
-            assertTrue(response.isPartial());
-            assertNotNull(response.getId());
+            assertTrue(submitResponse.isPartial());
+        }
+
+        GetAsyncSearchRequest getRequest = new GetAsyncSearchRequest(submitResponse.getId());
+        AsyncSearchResponse getResponse = highLevelClient().asyncSearch().getAsyncSearch(getRequest, RequestOptions.DEFAULT);
+        while (getResponse.isRunning()) {
+            getResponse = highLevelClient().asyncSearch().getAsyncSearch(getRequest, RequestOptions.DEFAULT);
         }
-    }
 
+        assertTrue(getResponse.getVersion() >= 0);
```

Comment:
watch out: we removed this from the response in https://github.com/elastic/elasticsearch/pull/53960

Reply:
Thanks, just see this while merging in master